tests/utils/test__select_column.py

```diff
@@ -307,64 +417,63 @@ def test_callable_string_methods(df1):
-    ) == list(df1.filter(regex=r"\d$").columns)
+    assert_index_equal(
+        _select_column_names(lambda x: x.name.endswith(("1", "2", "3")), df1),
+        df1.filter(regex=r"\d$").columns,
+    )
 
-    assert _select_column_names(lambda x: "d" in x.name, df1) == list(
-        df1.filter(regex="d").columns
+    assert_index_equal(
+        _select_column_names(lambda x: "d" in x.name, df1),
+        df1.filter(regex="d").columns,
     )
 
-    assert _select_column_names(
-        lambda x: x.name.startswith("code") and x.name.endswith("1"), df1
-    ) == list(df1.filter(regex=r"code.*1$").columns)
+    assert_index_equal(
+        _select_column_names(
+            lambda x: x.name.startswith("code") and x.name.endswith("1"), df1
+        ),
+        df1.filter(regex=r"code.*1$").columns,
+    )
 
-    assert _select_column_names(
-        lambda x: x.name.startswith("code") or x.name.endswith("1"), df1
-    ) == list(df1.filter(regex=r"^code.*|.*1$").columns)
+    assert_index_equal(
+        _select_column_names(
+            lambda x: x.name.startswith("code") or x.name.endswith("1"), df1
+        ),
+        df1.filter(regex=r"^code.*|.*1$").columns,
+    )
 
 
 def test_callable_computations(df1):
     """
     Test _select_column_names function on callables,
     specifically for computations.
     """
-    assert _select_column_names(lambda x: x.isna().any(), df1) == list(
-        df1.columns[df1.isna().any().array]
-    )
-
-
-def test_regex(df1):
-    """Test _select_column_names function on regular expressions."""
-    assert _select_column_names(re.compile(r"\d$"), df1) == list(
-        df1.filter(regex=r"\d$").columns
+    assert_index_equal(
+        _select_column_names(lambda x: x.isna().any(), df1),
+        df1.columns[df1.isna().any().array],
     )
 
 
-def test_tuple(df_tuple):
-    """Test _select_column_names function on tuple."""
-    assert _select_column_names(("A", "D"), df_tuple) == ("A", "D")
-
-
 def test_list_various(df1):
     """Test _select_column_names function on list type."""
 
     assert _select_column_names(["id", "Name"], df1) == ["id", "Name"]
     assert _select_column_names(["id", "code*"], df1) == list(
         df1.filter(regex="^id|^code").columns
     )
-    assert (
-        _select_column_names(["id", "code*", slice("code", "code2")], df1)
-        == df1.filter(regex="^(id|code)").columns.tolist()
-    )
+    assert [
+        *_select_column_names(["id", "code*", slice("code", "code2")], df1)
+    ] == df1.filter(regex="^(id|code)").columns.tolist()
     assert _select_column_names(["id", "Name"], df1) == ["id", "Name"]
 
 
 def test_list_boolean(df):
     """Test _select_column_names function on list of booleans."""
     booleans = [True, True, True, False, False, False, True, True, True]
-    assert _select_column_names(booleans, df) == list(df.columns[booleans])
+    assert_index_equal(
+        _select_column_names(booleans, df), df.columns[booleans]
+    )
```

Comment:
Is there a test missing for checking an error is raised when the list of booleans' length is not equal to the number of columns?

Reply:
yes ... `test_boolean_list_dtypes`  ... lines 324-336